observability/opentelemetry/src/main/java/org/wildfly/extension/opentelemetry/OpenTelemetrySubsystemRegistrar.java

```diff
@@ -158,18 +170,76 @@ class OpenTelemetrySubsystemDefinition extends PersistentResourceDefinition {
             .setRestartAllServices()
             .build();
 
-    public static final AttributeDefinition[] ATTRIBUTES = {
+    public static final List<AttributeDefinition> ATTRIBUTES = List.of(
             SERVICE_NAME, EXPORTER, ENDPOINT, SPAN_PROCESSOR_TYPE, BATCH_DELAY, MAX_QUEUE_SIZE, MAX_EXPORT_BATCH_SIZE,
             EXPORT_TIMEOUT, SAMPLER, RATIO
-    };
+    );
+
+    private Supplier<WildFlyOpenTelemetryConfig> opentelemetryConfig;
```

Comment:
If you want to feed your DUPs directly...
```java
private final AtomicReference<OpenTelemetryConfig> openTelemetryConfig = new AtomicReference<>();
```
Then, in your deployment chain contributor, create your OpenTelemetryDeploymentProcessor via:
```java
new OpenTelemetryDeploymentProcessor(this.openTelemetryConfig::get)
```

and then set the value of the atomic reference in your runtime handler.

Reply:
I don't have a strong preference. I just want to do it The Right Way. :)